src/Generator/Generators/CLI/CLITypePrinter.cs

```diff
@@ -146,11 +146,11 @@ public override TypePrinterResult VisitDelegate(FunctionType function)
             Enumeration @enum;
             if (pointee.TryGetEnum(out @enum))
             {
-                var typeName = @enum.Visit(this);
+                var typeName = VisitDeclaration(@enum, quals);
```

Comment:
What's the reason for this change?

Reply:
The generator does not generate fully qualified parameter name. For example instead of `CLI::EnumParam` it just generates `EnumParam`, which does not build.

Reply:
We should perhaps fix the real issue, that is, make visiting an enum generate the qualified name. @tritao what do you think?

Reply:
It does generate qualified name for other enum scenarios, does not seem to do it if enum is passed by ref, hence this fix.

Reply:
You should be able to do `@enum.Visit(this, quals);`.

Can you give that a try?

Reply:
There is no `Visit` overload that takes a second quals param.

Reply:
You're right, it's only supported for types, my bad.

Reply:
@IIFE Can you try removing the following method from `CLITypePrinter`?

```csharp
        public override TypePrinterResult VisitEnumDecl(Enumeration @enum)
        {
            return @enum.Name;
        }
``` 

That will call into the base `CppTypePrinter`, which should print enums using the general qualified declaration code.

Reply:
Removing that method causes StackOverflowException when generator is run.